util/testutil/dockerd/client/ping.go

```diff
@@ -0,0 +1,35 @@
+package client
+
+import (
+	"context"
+	"net/http"
+	"path"
+
+	"github.com/moby/buildkit/util/testutil/dockerd/client/types"
+)
+
+type PingResponse struct{}
+
+func (cli *Client) Ping(ctx context.Context) (types.Ping, error) {
+	var ping types.Ping
+
+	// Using cli.buildRequest() + cli.doRequest() instead of cli.sendRequest()
+	// because ping requests are used during API version negotiation, so we want
+	// to hit the non-versioned /_ping endpoint, not /v1.xx/_ping
+	req, err := cli.buildRequest(ctx, http.MethodHead, path.Join(cli.basePath, "/_ping"), nil, nil)
+	if err != nil {
+		return ping, err
+	}
+	serverResp, err := cli.doRequest(req)
```

Comment:
FWIW; you can probably get rid of the `serverResp` and maybe even the `checkResponseErr` as well; honestly that's a mess; also see https://github.com/moby/moby/pull/49420

Reply:
Done. I followed the linked PR and removed it in the same places.

Reply:
Thanks! On my phone right now, so probably not a good idea to look at the diff, but yeah; for the purpose of this test client, we probably don't need most of that (we can cleanup the "checkResponseError" later; I doubt it's of much use for tests here)